gnomad_qc/v5/sample_qc/generate_qc_mt.py

```diff
@@ -0,0 +1,271 @@
+"""
+Script to create a joint gnomAD v4 + AoU v8 dense MatrixTable filtered to predetermined QC sites.
+
+For more information about the predetermined QC sites, see:
+https://gnomad.broadinstitute.org/news/2023-11-gnomad-v4-0/#high-quality-sites-definition,
+https://github.com/broadinstitute/gnomad_qc/blob/main/gnomad_qc/v4/sample_qc/generate_qc_mt.py,
+and https://github.com/Nealelab/ccdg_qc/blob/master/scripts/pca_variant_filter.py.
+
+The joint MatrixTable created by this script is used for relatedness and genetic ancestry inference.
+"""
+
+import argparse
+import logging
+from typing import List
+
+import hail as hl
+
+from gnomad_qc.v4.resources import sample_qc as v4_sample_qc
+from gnomad_qc.v5.resources.basics import (
+    add_project_prefix_to_sample_collisions,
+    get_checkpoint_path,
+    get_logging_path,
+    get_samples_to_exclude,
+)
+from gnomad_qc.v5.resources.constants import AOU_WGS_BUCKET, WORKSPACE_BUCKET
+from gnomad_qc.v5.resources.meta import sample_id_collisions
+from gnomad_qc.v5.resources.sample_qc import get_joint_qc, hard_filtered_samples
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("generate_qc_mt")
+logger.setLevel(logging.INFO)
+
+
+def union_aou_mts(
+    s_to_exclude: hl.expr.SetExpression,
+    ht: hl.Table,
+    test: bool = False,
+    n_partitions: int = 10000,
+    entry_annotations: List[str] = ["GT"],
+    overwrite: bool = False,
+) -> None:
+    """
+    Filter AoU ACAF and exome MTs to QC MT sites, remove samples to exclude, and union MTs.
+
+    .. note::
+        Both AoU MTs have 145192 partitions.
+
+    :param s_to_exclude: Set of sample IDs to exclude from the AoU MatrixTable.
+    :param ht: Table containing the gnomAD QC sites.
+    :param test: If true, filter to the first 2 partitions for testing.
+        Default is False.
+    :param n_partitions: Number of desired partitions for the unioned MatrixTable.
+        Default is 10000.
+    :param entry_annotations: List of entry annotations to keep in the unioned MatrixTable.
+        Default is ["GT"].
+    :param overwrite: Whether to overwrite output data. Default is False.
+    :return: None; function writes unioned MT to temporary path.
+    """
+    logger.info(
+        "Loading AoU ACAF and exome MatrixTables and removing unnecessary annotations..."
+    )
+    acaf_mt = hl.read_matrix_table(
+        f"gs://{AOU_WGS_BUCKET}/acaf_threshold/splitMT/hail.mt"
+    ).select_entries(*entry_annotations)
+    exome_mt = hl.read_matrix_table(
+        f"gs://{AOU_WGS_BUCKET}/exome/splitMT/hail.mt"
+    ).select_entries(*entry_annotations)
+
+    def _filter_aou_mt(
+        mt: hl.MatrixTable,
+        s_to_exclude: hl.expr.SetExpression,
+        ht: hl.Table,
+        test: bool,
```

Comment:
Since these are always the same, just use the ones passed into the outer function